feat: per-stage claude -p orchestration for implement-bead (#28)
scotthamilton77/agents-config: src/plugins/beads/.claude/commands/implement-bead.md

# implement-bead

Invoke the implement-bead skill for the given bead ID.

> This slash command is the entry point for the shell driver's per-stage
> `claude -p` invocation. It takes a single argument — the bead ID — and

Author: ```suggestion
This slash command is the entry point bead implementation. It takes a single argument — the bead ID — and
```

Author: Acknowledged — the worker agent architecture (bead-implementor, implement-bead SKILL.md design, report contract) will be redesigned in a dedicated follow-up PR. This PR delivers the pipeline scaffolding; the worker layer redesign is next.

hands off to the implement-bead skill, which reads the appropriate stage
step-bead, dispatches subagents/skills per the stage spec, persists
state-out to bd/filesystem, and exits.

## Invocation

```
/implement-bead <bead-id>
```

Arguments: `$ARGUMENTS` receives the bead ID (e.g. `proj-42`).

## Behavior

1. Parse `$ARGUMENTS` as the bead ID.
2. Invoke the `implement-bead` skill with that bead ID.
3. The skill drives ONE stage of the bead's active molecule and exits.

## Supported invocation contexts

- **Interactive Claude session** — skill is discovered via the normal
  skill-lookup path; the session is interactive.
- **Shell driver `claude -p`** — the driver spawns `claude -p --session-id
  <uuidv5> "/implement-bead <bead-id>"` from the stage's cwd. Claude Code

>   resolves this as a globally-installed slash command (see below).

Author: How is this relevant to the execution of the command?

Author: Acknowledged — the worker agent architecture (bead-implementor, implement-bead SKILL.md design, report contract) will be redesigned in a dedicated follow-up PR. This PR delivers the pipeline scaffolding; the worker layer redesign is next.

## Slash-command resolution

This command is installed globally to `~/.claude/commands/` by the project's
`install.sh` installer (via the beads plugin overlay phase). It is therefore
available from any cwd — including worktree subdirectories — via Claude Code's
global command discovery, without any project-local `.claude/` directory or

> walk-up resolution required.

Author: Not relevant to the command.  The command is written for the agent, not a human, and should "command" - it should contain imperatives/instructions on what to do with zero irrelevant context.

Author: Acknowledged — the worker agent architecture (bead-implementor, implement-bead SKILL.md design, report contract) will be redesigned in a dedicated follow-up PR. This PR delivers the pipeline scaffolding; the worker layer redesign is next.

> walk-up resolution required.

Author: Fixed in e038994. The architecture spec (docs/specs/bead-pipeline-architecture.md §5.4) now states that /implement-bead is installed globally to ~/.claude/commands/ by install.sh and is discoverable from any cwd via Claude Code's global command lookup — not via project-local walk-up. The command file already had the correct description; the spec was the stale one.

## Notes

- The driver sets cwd before spawning each `claude -p` invocation per the

>   cwd contract in the architecture spec (section 5.4).

Author: Fixed in 784c4820d7c8330c0b37f037f6894ba0286c5429. Updated doc to state the command is globally installed to ~/.claude/commands/ by install.sh, discoverable from any cwd.
